src/lib/components/selectv2/Selectv2.component.js

```diff
@@ -114,21 +119,69 @@ const Menu = (props) => {
   return <components.Menu {...props} />;
 };
 
-const MenuList = (props) => {
-  const { options, children, maxHeight, getValue } = props;
+const getScrollOffset = (
+  list,
+  index: number,
+  itemCount: number,
+  offset: number,
+): number => {
+  const { itemSize, height } = list.props;
+  const scrollOffset = list.state ? list.state.scrollOffset : 0;
+  const lastItemOffset = Math.max(0, itemCount * itemSize - height);
+  const maxOffset = Math.min(lastItemOffset, index * itemSize);
+  const minOffset = Math.max(0, index * itemSize - height + itemSize);
+  if (scrollOffset >= minOffset && scrollOffset <= maxOffset) {
+    return scrollOffset;
+  } else if (scrollOffset < minOffset) {
+    return minOffset === 0 ? minOffset : minOffset + offset;
+  } else {
+    return maxOffset === 0 ? maxOffset : maxOffset - offset;
+  }
+};
 
-  if (children.length > 4) {
-    const [value] = getValue();
-    const height = convertRemToPixels(
+const MenuList = (props) => {
+  const listRef = useRef();
+  const { children, getValue } = props;
+  const [selectedOption] = getValue();
+  const optionHeight =
+    convertRemToPixels(
       parseFloat(props.selectProps.isDefault ? spacing.sp32 : spacing.sp24),
     ) || 32;
-    const initialOffset = options.indexOf(value) * height - 3 * height;
+  let selectedIndex = 0;
+  let focusedIndex = 0;
+
+  if (children && children.length > 0) {
+    selectedIndex = children.findIndex(
+      (child) => child.props.data === selectedOption,
+    );
+    focusedIndex = props.focusedOption
+      ? children.findIndex((child) => child.props.data === props.focusedOption)
+      : selectedIndex;
+  }
+
+  const initialOffset = selectedIndex * optionHeight - 3 * optionHeight;
+
+  useEffect(() => {
+    if (listRef && listRef.current) {
+      listRef.current.scrollTo(
+        getScrollOffset(
+          listRef.current,
+          focusedIndex,
+          children.length,
+          optionHeight / 4,
+        ),
+      );
+    }
+  }, [children.length, focusedIndex, optionHeight]);
+
+  if (children.length > 4) {
```

Comment:
I would extract "4" in a variable `ITEMS_PER_SCROLL_WINDOW` and derive `initialOffset`, `optionHeight / 4` and `height={optionHeight * 4.5}` from this variable.